src/ipc/ipc4/logging.c

```diff
@@ -59,21 +65,28 @@ struct task mtrace_task;
 
 static void mtrace_log_hook(size_t written, size_t space_left)
 {
-	if (arch_proc_id() == MTRACE_IPC_CORE &&
-	    (space_left < NOTIFY_BUFFER_STATUS_THRESHOLD ||
-	     k_uptime_get() - mtrace_notify_last_sent >= mtrace_aging_timer)) {
+	uint64_t delta;
+
+	mtrace_bytes_pending += written;
+
+	/* TODO: if hook is called on non-zero core, logs
+	 * maybe lost with too slow aging timer. need to
+	 * figure out a safe way to wake up the mtrace task
+	 * from another core.
+	 */
+	if (arch_proc_id() != MTRACE_IPC_CORE)
+		return;
+
+	delta = k_uptime_delta(&mtrace_notify_last_sent);
```

Comment:
here updating `mtrace_notify_last_sent` is intended?

Reply:
Damn, rebasing brought it back. Will address.